csrc/preseg_passes/remove_bcast_squeeze.cpp

```diff
@@ -318,13 +339,79 @@ TensorView* maybeDoReplacement(TensorView* orig) {
   if (!isReplaceableExpr(second)) {
     return orig;
   }
+  AxisOps second_ops = exprToAxisOps(second);
```

Comment:
Having a hard time to understand what this function (`maybeDoReplacement`) is doing. What is the parameter assumed to be? What is supposed to be returned?

Reply:
I think `maybeDoReplacement` is trying to merge `tv->first->second->orig` as a `tv->merged->new_out` when both first and second are replaceable exprs.

i.e. when we have `tv->broadcast->squeeze`, we might be able to just cancel the two and ended up returning a `tv` directly.

The function returns the `new_out` after the replay. The logic here is that:
if the returned pointer is different from `orig`, it would consider a replacement has happened and would try to the same loop with `new_out`;
if the returned pointer is the same as `orig`, merge failed, it would skip `second` here and move on and push inputs to `second` as new candidate as `orig` in the stack.

Reply:
So the added logic here is, when we try to swap `tv->first->second->orig` as `tv->replayed_second->replayed_first`, we return `replayed_second->output(0)`.

Even though we are not merging two consecutive replaceable operations, by returning `replayed_second->output(0)` instead of `orig`, we kept `replayed_second` as the candidate for the iteration, effectively skipped unary-op `first` from preventing us merging neighboring replaceable operations.